src/SonarScanner.MSBuild.PreProcessor/WebServer/SonarCloudWebServer.cs

```diff
@@ -118,6 +111,23 @@ public override async Task<IList<SensorCacheEntry>> DownloadCache(ProcessedArgs
             }
         }
 
+        // Do not use the downloaders here, as this is an unauthenticated request
+        public override async Task<Stream> DownloadJreAsync(JreMetadata metadata)
```

Comment:
This and the other one on SonarQubeWebServer are the callback to be passed to the cache.
It is the only point of separation between SQ and SC, and the cache is supposed to handle failures